src/api/services/EscrowService.ts

```diff
@@ -174,6 +182,59 @@ export class EscrowService {
         await this.escrowRepo.destroy(id);
     }
 
+    @validate()
+    public async lock( @request(EscrowLockRequest) data: any): Promise<void> {
+        // fetch the escrow
+        const escrow = await this.findOne(data.escrowId, false);
+        // fetch the address
+        const address = await this.addressService.findOne(data.addressId, false);
+        // escrowfactory to generate the lockmessage
+        const messageInput = {
+            escrow: escrow.toJSON(),
+            address: address.toJSON(),
+            listing: data.itemHash,
+            nonce: data.nonce,
+            memo: data.memo,
+            action: '',
+            item: ''
+        };
+        const escrowActionMessage = await this.escrowFactory.getLockMessage(messageInput);
+        return await this.messageBroadcastService.broadcast();
+    }
```

Comment:
you are creating somekind of messageinput and then passing it to generate a message for you using a factory?
whats the point of using the factory here then, you could just manually create the escrowActionMessage?

Pass the validated EscrowLockRequest to this.escrowFactory.get() and broadcast the result.

data is typed as any, should be EscrowLockRequest.

Reply:
same things to refund and release